src/main.ts

```diff
@@ -50,12 +50,12 @@ export async function run() {
         core.info('Setting GOROOT for Go version < 1.9');
         core.exportVariable('GOROOT', installDir);
       }
-
-      const added = await addBinToPath();
-      core.debug(`add bin ${added}`);
-      core.info(`Successfully set up Go version ${versionSpec}`);
     }
 
+    const added = await addBinToPath();
+    core.debug(`add bin ${added}`);
+    core.info(`Successfully set up Go version ${versionSpec}`);
```

Comment:
I think we should leave `core.info` in the `if` statement because it can be empty if no version is specified.

Reply:
But can we say that we have successfully installed go before the `go/bin` was added to the path ? I think we have to wrap 57 line in condition depends on version spec is set or not. If set - we do installed the version, If not we just added precached version to the path - no need to log anything.
```suggestion
    if (versionSpec) {
      core.info(`Successfully set up Go version ${versionSpec}`)
    };
```